src/cargo/util/command_prelude.rs

```diff
@@ -1106,6 +1111,81 @@ pub fn get_registry_candidates() -> CargoResult<Vec<clap_complete::CompletionCan
     }
 }
 
+fn get_feature_candidates() -> CargoResult<Vec<clap_complete::CompletionCandidate>> {
+    let gctx = new_gctx_for_completions()?;
+
+    let manifest_path = match find_root_manifest_for_wd(gctx.cwd()) {
+        Ok(path) => path,
+        Err(_) => return Ok(Vec::new()),
+    };
+
+    let ws = match Workspace::new(&manifest_path, &gctx) {
+        Ok(ws) => ws,
+        Err(_) => return Ok(Vec::new()),
+    };
+
+    let current_dir = std::env::current_dir()?;
+
+    let current_pkg = ws
+        .members()
+        .find(|pkg| current_dir.starts_with(pkg.root()))
+        .or_else(|| ws.members().next())
+        .cloned();
```

Comment:
For `--bin [TAB]`, we complete for any package in the workspace in case `--package` is used